ORBIT/manager.py

```diff
@@ -1438,38 +1509,206 @@ def overnight_capex(self):
         """
-        Returns project soft costs per kW. Default numbers are based on the
-        Cost of Energy Review (Stehly and Beiter 2018).
+        Returns the construction insurance capital cost of the project.
+        Methodology from ORCA model, default values used in 2022 Cost of Wind Energy Review.
         """
 
-        insurance = self.project_params.get("construction_insurance", 44)
-        financing = self.project_params.get("construction_financing", 183)
-        contingency = self.project_params.get("contingency", 316)
-        commissioning = self.project_params.get("commissioning", 44)
-        decommissioning = self.project_params.get("decommissioning", 58)
+        try:
+            construction_insurance_per_kW = self.config["project_parameters"]["construction_insurance"]
+            num_turbines = self.config["plant"]["num_turbines"]
+            rating = self.config["turbine"]["turbine_rating"]
+            construction_insurance = construction_insurance_per_kW * num_turbines * rating * 1000
+            
+        except:
+            contruction_insurance_factor =  self.project_params.get("construction_insurance_factor", 0.0115)
+            construction_insurance = (self.turbine_capex + self.bos_capex + self.project_capex) *\
+                                    contruction_insurance_factor
+
+        return construction_insurance
+
+    @property
+    def decommissioning_capex(self):
+        """
+        Returns the decommissioning capital cost of the project.
+        Methodology from ORCA model, default values used in 2022 Cost of Wind Energy Review.
+        """
+
+        try:
+            decommissioning_per_kW = self.config["project_parameters"]["decommissioning"]
+            num_turbines = self.config["plant"]["num_turbines"]
+            rating = self.config["turbine"]["turbine_rating"]
+            decommissioning = decommissioning_per_kW * num_turbines * rating * 1000
+            
+        except:
+            decommissioning_factor =  self.project_params.get("decommissioning_factor", 0.175)
+            decommissioning = self.installation_capex * decommissioning_factor
+
+        return decommissioning
+
+    @property
+    def project_completion_capex(self):
+        """
+        Returns the project completion capital cost of the project.
+        Methodology from ORCA model, default values used in 2022 Cost of Wind Energy Review.
+        """
+
+        try:
+            project_completion_per_kW = self.config["project_parameters"]["project_completion"]
+            num_turbines = self.config["plant"]["num_turbines"]
+            rating = self.config["turbine"]["turbine_rating"]
+            project_completion = project_completion_per_kW * num_turbines * rating * 1000
+            
+        except:
+            project_completion_factor =  self.project_params.get("project_completion_factor", 0.0115)
+            project_completion = (self.turbine_capex + self.bos_capex + self.project_capex) *\
```

Comment:
is there a term for turbine_capex + bos_capex + project_capex? Just wondering if you can't save some space by making another property like `total_capex` but only include these three components. Just a thought, but you don't have to...

Reply:
I don't know of a good economics term for turbine_capex + bos_capex + project_capex, so I will keep it as it is.